src/contracts/GPv2Settlement.sol

```diff
@@ -69,6 +69,9 @@ contract GPv2Settlement {
         bytes orderUid
     );
 
+    /// @dev Event emitted when an order is invalidated.
+    event OrderInvalidated(address indexed owner, bytes orderUid);
```

Comment:
Since this function isn't called in `settle` and even then only rarely, I would make both parameters indexed for ease of searching if a specific order was invalidated. 
```suggestion
    event OrderInvalidated(address indexed owner, bytes indexed orderUid);
```

Reply:
Unfortunately, this causes the event to only be emitted with 2 topics, the second one being `keccak256(orderUid)` making it not possible to recover the actual UI from just the log data.

Another option would be
```solidity
event OrderInvalidated(
    address indexed owner,
    bytes32 indexed orderKey, // = keccak256(orderUid)
    bytes orderUid,
);
```

Reply:
I see, then to avoid confusion I'd keep it like you did it in the first place, just to avoid introducing the concept of an `orderKey`. I imagine that every time we check if an order is invalidated we are going to know who the owner is, and then there shouldn't be so many other invalidated orders that it's hard to search through them.